firebase-firestore/src/test/java/com/google/firebase/firestore/integration/FirebaseFirestoreTest.java

```diff
@@ -0,0 +1,219 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.integration;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.firebase.firestore.testutil.TestUtil.map;
+import static com.google.firebase.firestore.util.Executors.BACKGROUND_EXECUTOR;
+import static org.mockito.Mockito.verify;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreIntegrationTestFactory;
+import com.google.firebase.firestore.UserDataReader;
+import com.google.firebase.firestore.core.UserData;
+import com.google.firebase.firestore.model.DatabaseId;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firebase.firestore.model.mutation.FieldTransform;
+import com.google.firebase.firestore.model.mutation.Precondition;
+import com.google.firebase.firestore.model.mutation.SetMutation;
+import com.google.firebase.firestore.remote.RemoteSerializer;
+import com.google.firestore.v1.Write;
+import com.google.firestore.v1.WriteRequest;
+import com.google.firestore.v1.WriteResponse;
+import com.google.firestore.v1.WriteResult;
+
+import org.bouncycastle.util.Arrays;
```

Comment:
Is this a new dependency? How does it work without gradle change?

Reply:
Hmm ... I thought this was from Guava. I will refactor to not use this.